runtime/lua/vim/lsp/util.lua

```diff
@@ -440,7 +422,24 @@ function M.apply_text_edits(text_edits, bufnr, offset_encoding)
 
   -- Apply text edits.
   local is_cursor_fixed = false
+  local has_eol_text_edit = false
   for _, text_edit in ipairs(text_edits) do
+    -- Some LSP servers may return +1 range of the buffer content but nvim_buf_set_text can't accept it so we should fix it here.
+    local max = vim.api.nvim_buf_line_count(bufnr)
+    local len = _str_utfindex_enc(vim.api.nvim_buf_get_lines(bufnr, -2, -1, false)[1] or '', nil, offset_encoding)
```

Comment:
Can we first convert the text edit character to bytes, and then re-use that on line 446 and 448?